axon-framework/events/event-processors/README.md

```diff
@@ -253,12 +308,213 @@ public interface ErrorHandler {
+
+    private final EventProcessingConfiguration eventProcessingConfiguration;
+
+    public DeadletterProcessor(EventProcessingConfiguration eventProcessingConfiguration) {
+        this.eventProcessingConfiguration = eventProcessingConfiguration;
+    }
+
+    public void repairAnyEvent() {
+        eventProcessingConfiguration.sequencedDeadLetterProcessor(PROCESSING_GROUP)
+                                    .ifPresent(SequencedDeadLetterProcessor::processAny);
+    }
+}
+```
+
+{% endtab %}
+
+{% tab title="Process all dead-letters in the queue" %}
+
+```java
+public class DeadletterProcessor {
+
+    private final EventProcessingConfiguration eventProcessingConfiguration;
+
+    public DeadletterProcessor(EventProcessingConfiguration eventProcessingConfiguration) {
+        this.eventProcessingConfiguration = eventProcessingConfiguration;
+    }
+
+    public void repairAllEvents() {
+        Optional<Iterable<Iterable<DeadLetter<? extends EventMessage<?>>>>> deadLetters = eventProcessingConfiguration.deadLetterQueue(
+                PROCESSING_GROUP).map(SequencedDeadLetterQueue::deadLetters);
+
+        deadLetters.ifPresent(sequences -> {
+            // Iterate over all sequences
+            for (Iterable<DeadLetter<? extends EventMessage<?>>> sequence : sequences) {
+                // Iterate over all dead-letters belonging to the same sequence
+                for (DeadLetter<? extends EventMessage<?>> deadLetterInSequence : sequence) {
+                    eventProcessingConfiguration.sequencedDeadLetterProcessor(PROCESSING_GROUP).ifPresent(
+                            eventMessageSequencedDeadLetterProcessor -> {
+                                eventMessageSequencedDeadLetterProcessor.process(
+                                        deadLetter -> deadLetter.message().getIdentifier()
+                                                                .equals(deadLetterInSequence.message()
+                                                                                            .getIdentifier()));
+                            });
+                }
+            }
+        });
+    }
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+You can implement a custom dead-letter policy to exclude some events from the dead-letter queue, these events will be
+skipped. This policy is not only called for initial failures but also when dead-lettered events are processed
+unsuccessfully again.
+
+```java
+@Configuration
+public class CustomDeadLetterPolicy {
+
+    @Autowired
+    public void configure(EventProcessingConfigurer configurer) {
+        configurer.registerDeadLetterPolicy(PROCESSING_GROUP, configuration ->
+                (letter, cause) -> {
+                    if (cause instanceof NullPointerException) {
+                        // It's pointless..
+                        return Decisions.doNotEnqueue();
+                    }
+                    final int retries = (int) letter.diagnostics().getOrDefault("retries", -1);
```

Comment:
The `diagnostics` bit makes me wonder whether we should have another subsection about what the `DeadLetter` stores.
The section would be short and simple I think, essentially a version of the API docs of the `DeadLetter`.
Thus, a bullet-list of the fields and what there intent is. 

That way, we do not have to explain diagnostics in this section, as this section should be focused around how to make that decision.